File: Backend/src/main/java/com/naoufalayache/authentication/services/MessageService.java
package com.naoufalayache.authentication.services;

import java.util.List;
import java.util.Optional;

import com.naoufalayache.DTO.MessageDTO;
import com.naoufalayache.DTO.ResponseDTO;
import com.naoufalayache.authentication.model.Message;

public interface MessageService {
    public ResponseDTO upsert(Message message);
    public Optional<Message> findById(Long id);
    public ResponseDTO delete(Message message);
    List<MessageDTO> getMessages(Long conversationId, int page, int size);
}

File: Frontend/src/app/services/message.service.ts
import { Injectable } from '@angular/core';
import { ConfigService } from './config.service';
import { HttpClient, HttpParams } from '@angular/common/http';
import { DeleteConv, Message, MessageSended, Response } from '../models/model';

@Injectable({
  providedIn: 'root',
})
export class MessageService {
  constructor(
    private http: HttpClient,
    private configService: ConfigService,
  ) {}

  upsertMessages(request: MessageSended) {
    return this.http.post<Response>(`${this.configService.apiUrl}/message/upsert`, request);
  }

  getMessages(conversationId: number, page: number, size: number) {
    const params = new HttpParams()
      .set('conversationId', conversationId)
      .set('page', page)
      .set('size', size);
      return this.http.get<Message[]>(`${this.configService.apiUrl}/message`,{params})
  }

  deleteMessage(request: DeleteConv){
    return this.http.delete<Response>(`${this.configService.apiUrl}/message`, {body : request})
  }
}
